routes/login.js

```diff
@@ -9,11 +10,41 @@ router.get("/", (req, res) => {
         return;
     }
     res.render('login/login');
-})
+});
 
-router.post("/", passport.authenticate('local', {failureRedirect: '/login', failureFlash: true}),(req, res) => {
-    //do passport stuff here
-    res.redirect("/class");
-})
+router.post("/", (req, res, next) => {
+    passport.authenticate('local', (err, user, info) => {
+        if (err) {
+            return next(err);
+        }
+        return db.accountLocked(req.body.username).then((lockData) => {
+            if (lockData.locked) {
+                let date = new Date(lockData.lockAccountUntil);
+                let data = {
+                    hasErrors: true,
+                    errors: [`${lockData.failedLoginAttempts} failed login attempts. Your account has been locked until ${date}`]
+                };
+                return res.render('login/login', data);
+            }
+            if (!user) {
+                // Login failed
+                return db.failedLoginAttempt(req.body.username, lockData.failedLoginAttempts).then(() => {
+                    return res.redirect('/login');
```

Comment:
should we send the user an error message that there login failed

Reply:
I send the error iff their actual attempt was blocked. You will not see the error message even if you fail the attempt, but enough time has passed for it not to be denied. I tried showing the message every time the user failed the login but it looked a bit obnoxious.